diracx-cli/tests/test_jobs.py

```diff
@@ -1,13 +1,128 @@
 from __future__ import annotations
 
 import json
+import os
+import tempfile
+
+import pytest
+from pytest import raises
 
 from diracx import cli
+from diracx.core.models import ScalarSearchSpec
+from diracx.core.preferences import get_diracx_preferences
 
+TEST_JDL = """
+    Arguments = "jobDescription.xml -o LogLevel=INFO";
+    Executable = "dirac-jobexec";
+    JobGroup = jobGroup;
+    JobName = jobName;
+    JobType = User;
+    LogLevel = INFO;
+    OutputSandbox =
+        {
+            Script1_CodeOutput.log,
+            std.err,
+            std.out
+        };
+    Priority = 1;
+    Site = ANY;
+    StdError = std.err;
+    StdOutput = std.out;
+"""
+
+
+@pytest.fixture
+async def jdl_file():
+    with tempfile.NamedTemporaryFile(
+        mode="w", encoding="utf-8", delete=False
+    ) as temp_file:
+        temp_file.write(TEST_JDL)
+        temp_file.flush()
+        jdl_filename = temp_file.name
+
+    yield jdl_filename
+
+    os.remove(jdl_filename)
```

Comment:
I think this can be simplified to one of these:

```suggestion
@pytest.fixture
async def jdl_file():
    with tempfile.NamedTemporaryFile(mode="w", encoding="utf-8") as temp_file:
        temp_file.write(TEST_JDL)
        temp_file.flush()
        yield temp_file.name
```

else, if using `delete=False` the flush is pointless:

```suggestion
@pytest.fixture
async def jdl_file():
    with tempfile.NamedTemporaryFile(
        mode="w", encoding="utf-8", delete=False
    ) as temp_file:
        temp_file.write(TEST_JDL)

    yield temp_file.name

    os.remove(temp_file.name)
```